watchtower/wtclient/backup_task.go

```diff
@@ -0,0 +1,344 @@
+
+	// Parse the non-dust outputs from the breach transaction,
+	// simultaneously computing the total amount contained in the inputs
+	// present. We can't compute the exact output values at this time
+	// since the task has not been assigned to a session, at which point
+	// parameters such as fee rate, number of outputs, and reward rate will
+	// be finalized.
+	var (
+		totalAmt      int64
+		toLocalInput  input.Input
+		toRemoteInput input.Input
+	)
+
+	// If the to-local output is non-dust, add it to the set of candidate
+	// inputs and total input amount.
+	if breachInfo.RemoteOutputSignDesc != nil {
+		toLocalInput = input.NewBaseInput(
+			&breachInfo.RemoteOutpoint,
+			input.CommitmentRevoke,
+			breachInfo.RemoteOutputSignDesc,
+			0,
+		)
+		totalAmt += breachInfo.RemoteOutputSignDesc.Output.Value
+	}
+
+	// If the to-remote output is non-dust, add it to the set of candidate
+	// inputs and total input amount.
+	if breachInfo.LocalOutputSignDesc != nil {
+		toRemoteInput = input.NewBaseInput(
+			&breachInfo.LocalOutpoint,
+			input.CommitmentNoDelay,
+			breachInfo.LocalOutputSignDesc,
+			0,
+		)
+		totalAmt += breachInfo.LocalOutputSignDesc.Output.Value
+	}
+
+	return &backupTask{
+		chanID:        *chanID,
+		commitHeight:  breachInfo.RevokedStateNum,
+		breachInfo:    breachInfo,
+		toLocalInput:  toLocalInput,
+		toRemoteInput: toRemoteInput,
+		totalAmt:      btcutil.Amount(totalAmt),
+	}
+}
+
+// inputs returns all non-dust inputs that we will attempt to spend from.
+//
+// NOTE: Ordering of the inputs is not critical as we sort the transaction with
+// BIP69.
+func (t *backupTask) inputs() map[wire.OutPoint]input.Input {
+	inputs := make(map[wire.OutPoint]input.Input)
+	if t.toLocalInput != nil {
+		inputs[*t.toLocalInput.OutPoint()] = t.toLocalInput
+	}
+	if t.toRemoteInput != nil {
+		inputs[*t.toRemoteInput.OutPoint()] = t.toRemoteInput
+	}
+	return inputs
+}
+
+// bindSession determines if the backupTask is compatible with the passed
+// SessionInfo's policy. If no error is returned, the task has been bound to the
+// session and can be queued to upload to the tower. Otherwise, the bind failed
+// and should be rescheduled with a different session.
+func (t *backupTask) bindSession(session *wtdb.SessionInfo) error {
+	// First we'll begin by deriving a weight estimate for the justice
+	// transaction. The final weight can be different depending on whether
+	// the watchtower is taking a reward.
+	var weightEstimate input.TxWeightEstimator
+
+	// All justice transactions have a p2wkh output paying to the victim.
+	weightEstimate.AddP2WKHOutput()
+
+	// Next, add the contribution from the inputs that are present on this
+	// breach transaction.
+	if t.toLocalInput != nil {
+		weightEstimate.AddWitnessInput(input.ToLocalPenaltyWitnessSize)
+	}
+	if t.toRemoteInput != nil {
+		weightEstimate.AddWitnessInput(input.P2WKHWitnessSize)
+	}
+
+	// Now, compute the output values depending on whether FlagReward is set
```

Comment:
Carried over from the other PR, I think this can use that utility function referenced.